src/components/Identity/createComponent.js

```diff
@@ -17,135 +17,135 @@ const addEcidToPayload = (payload, ecid) => {
-              if (
-                config.thirdPartyCookiesEnabled &&
-                areThirdPartyCookiesSupportedByDefault(getBrowser(window))
-              ) {
-                payload.useIdThirdPartyDomain();
-              }
-            }
-          }
-
-          customerIds.addToPayload(payload);
-          return promise;
-        });
+      onBeforeDataCollectionRequest({ payload }) {
+        return accommodateIdentityOnRequest(payload);
+      },
+      onBeforeConsentRequest({ payload }) {
+        return accommodateIdentityOnRequest(payload);
       },
-
-      // Waiting for opt-in because we'll be reading the ECID from a cookie
       onResponse({ response }) {
-        return optIn.whenOptedIn().then(() => {
-          if (idMigrationEnabled) {
-            const identityResultPayloads = response.getPayloadsByType(
-              "identity:result"
-            );
-
-            const ecidPayload = find(
-              identityResultPayloads,
-              payload =>
-                payload.namespace && payload.namespace.code === ecidNamespace
-            );
-
-            if (ecidPayload) {
-              migration.createLegacyCookie(ecidPayload.id);
-            }
-          }
+        const promises = [];
+        if (idMigrationEnabled) {
+          const identityResultPayloads = response.getPayloadsByType(
+            "identity:result"
+          );
 
-          // If we were queuing requests until we received the identity cookie,
-          // and now we have the identity cookie, we can let the queued
-          // requests go out. Technically, we should always have an identity
-          // cookie at this point, but we check just to be sure.
-          if (deferredForEcid && hasIdentityCookie()) {
-            deferredForEcid.resolve();
+          const ecidPayload = find(
+            identityResultPayloads,
+            payload =>
+              payload.namespace && payload.namespace.code === ecidNamespace
+          );
+
+          if (ecidPayload) {
+            promises.push(migration.createLegacyCookie(ecidPayload.value));
           }
+        }
+
+        // If we were queuing requests until we received the identity cookie,
+        // and now we have the identity cookie, we can let the queued
+        // requests go out. Technically, we should always have an identity
+        // cookie at this point, but we check just to be sure.
+        if (deferredForEcid && hasIdentityCookie()) {
+          deferredForEcid.resolve();
+        }
 
-          return processIdSyncs(
+        promises.push(
+          processIdSyncs(
             response.getPayloadsByType("identity:exchange"),
             logger
-          );
-        });
+          )
+        );
+
+        return Promise.all(promises);
       }
     },
     commands: {
       setCustomerIds(options) {
-        return optIn
-          .whenOptedIn()
-          .then(() => customerIds.sync(options, logger));
```

Comment:
Same thing here. I pushed the logic that waits for consent down into `customerIds.sync`.